eth/stagedsync/stage_execute.go

```diff
@@ -384,8 +384,9 @@ func SpawnExecuteBlocksStage(s *StageState, u Unwinder, txc wrap.TxContainer, to
 	}
 
 	quit := ctx.Done()
-	useExternalTx := txc.Tx != nil
-	if !useExternalTx {
+	useInternalTx := txc.Tx == nil
```

Comment:
I suggest to restore usage of `useExternalTx` here because it's a kind of pattern used in all Erigon stages:
```go
useExternalTx := tx != nil
if !useExternalTx {
    var err error
    tx, err = cfg.db.BeginRw(context.Background())
    if err != nil {
        return err
    }
    defer tx.Rollback()
}
```

Reply:
done